### DISCUSSION TITLE
Multiple consoles

### DISCUSSION THREAD

**Discussion by jaril** [RecordReplay/devtools#6105] [General]
Started talking about ideas of how to address the feedback of "There's too much clutter in the console", and I brought up multiple consoles. Documenting the notes here:
- It might be helpful if console A/B/C were to keep their own filter/logpoint state so you can focus on a single idea for each console instead of muddling it up.
- Another way to approach this is you were able to "pin" a console message from global console to console A so that console A is a clean view
- Mental model — is it that we have multiple consoles? Or is it that we have different views of the global console?

**Comment by jaril** [comment]
Accidentally ran into this use case and used a workaround without realizing that it's similar.

I had 5+ print statements, but #1/2/3 were telling one story, and #4/5 were telling another. And it was useful to see that story in the console separately.

Absent multiple consoles, I just added a string to filter by in the log message, so I can use the console's filter to switch "views". e.g. #1/2/3 had `"story1", …`, then #4/5 had `"story2", …`.

**Reply by jaril** [reply]
Worked pretty dang good tbh

**Comment by jaril** [comment]
0 days since somebody's brought this up — capturing raw notes:

```
jcmorrow — Today at 4:25 AM
I've also thought this before 👍  I think personally I would opt for turning the numerator and denominator of our hit counter into inputs, much like we do with focus mode first, but I think it's helpful to have some way of doing this
Also on the topic of unicorn console - I really feel like adding highlight to some console messages is not the way to help people silence/pick out particular messages. I would much rather have one console per thing printing, which I can combine if I want, but maybe that's just me
brianvaughn — Today at 6:02 AM
That...is something I have not considered before! Sounds like a really interesting idea to explore.
TBH most of the time I'd want more than one per console for sure, but I can imagine potentially wanting multiple consoles (each with their own distinct flows). 
jcmorrow — Today at 6:04 AM
yeah, I like the way that natto.dev and other visual tools think of function outputs kind of like modular synth outputs, that can be patched to other things, and I feel like the truly futuristic take on things like devtools looks more like that than like the old school devtools of Chrome and FireFox. But familiarity is also important 😄
jlast — Today at 7:08 AM
Absolutely - this feels like something we should promote to a GH discussion with some mockups
```

**Comment by jonbell-lot23** [comment]
Hello! This issue isn't exactly the same, but it's in the same general space: Unified View Exploration.

[Here's the discussion in Discord](https://discord.com/channels/779097926135054346/976232715831615519)
[Here's a Loom walkthrough](https://www.loom.com/share/20adda09b4c346059d7de22ed53b3abf)
